src/trustedtypes.js

```diff
@@ -45,31 +50,52 @@ export const TrustedTypes = (function() {
   const privates = function(obj) {
     let v = privateMap.get(obj);
     if (v === undefined) {
-      v = Object.create({}); // initialize the private storage.
+      v = create(null); // initialize the private storage.
       privateMap.set(obj, v);
     }
     return v;
   };
 
+  /**
+   * Called before attacker-controlled code on an internal collections,
+   * copies prototype members onto the instance directly, so that later
+   * changes to prototypes cannot expose collection internals.
+   * @param {!T} collection
+   * @return {!T} collection
+   * @template T
+   */
+  function selfContained(collection) {
+    const proto = getPrototypeOf(collection);
+    if (proto == null || getPrototypeOf(proto) !== ObjectPrototype) {
+      throw new Error(); // Loop below is insufficient.
+    }
+    for (let key of getOwnPropertyNames(proto)) {
+      defineProperty(collection, key, {value: collection[key]});
+    }
+    return collection;
+  }
+
   /**
    * Map for private properties of Trusted Types object.
    * This is so that the access to the type constructor does not give
    * the ability to create typed values.
    * @type {WeakMap}
    */
-  const privateMap = new WeakMap();
+  const privateMap = selfContained(new WeakMap());
 
   /**
    * List of all configured policy names.
    * @type {Array<string>}
    */
-  const policyNames = [];
+  const policyNames = selfContained([]);
 
   /**
    * Map of all exposed policies, keyed by policy name.
    * @type {Map<string,Object>}
    */
-  const exposedPolicies = new Map();
+  const exposedPolicies = selfContained(new Map());
+
+  // TODO(msamuel): common base class for Trusted*?
```

Comment:
Go for it, not necessarily in this PR (or here, it doesn't really matter).

Reply:
Done here.